internal/core/src/exec/expression/UnaryExpr.cpp

```diff
@@ -1133,6 +1130,11 @@ PhyUnaryRangeFilterExpr::CanUseIndex() {
 
 bool
 PhyUnaryRangeFilterExpr::CanUseIndexForJson(DataType val_type) {
+    if (val_type != DataType::BOOL && val_type != DataType::INT64 &&
+        val_type != DataType::DOUBLE && val_type != DataType::STRING) {
```

Comment:
not supported array is by the nature of inverted index or a temporary solution.
I remember elastic can do array(there are some issues with nested arrays)

Reply:
> not supported array is by the nature of inverted index or a temporary solution. I remember elastic can do array(there are some issues with nested arrays)

a temporary solution. we will support array soon.